### ISSUE TITLE
[edge compute] - Update the agent to support Edge stacks

### ISSUE THREAD

**Issue by deviantony** [portainer/agent#115]
The Edge agent must be able to retrieve Edge stack related information as part of the periodic short poll request.

The agent must be able to store information related to Edge stacks (identifier, version number) in memory.

It must also be able to retrieve a stack file associated to an Edge stack and apply it.

If a stack is referenced in the memory of the agent but not exposed via short the short poll request anymore, the stack must be deleted.

**Comment by deviantony** [comment]
Blocking #116

**Comment by chiptus** [comment]
- rename edge stack manager to docker stack service
- edge stack manager (inside http package?)
  - contains a map stackeID -> version, stack file location, prune, status (pending, done, error)
  - when a stack is received, the manager will check its version vs the one in the map, and will deploy if needed
  - before deploying, stack object will be updated with the file, version, (prune if needed) and replace status to pending.
  - there will be a coroutine, that checks if there's a pending stack and deploy it.
  - if a stack is in error state (i.e. deploy failed), the stack will wait for redeployment until a new version is created (here we assume that the error happened because of an invalid config file)

**Comment by chiptus** [comment]
```
type EdgeStackVersion struct {
  ID
  Version
}


type EdgeStack struct {
  clone of portainer.EdgeStack
}

type EdgeStackManager interface {
  CompareStacks(EdgeStackVersion[]) error
}

// add to APIClient:
type APIClient interface {
   GetEdgeStackFile(EdgeStackID) (string, error)
   SetEdgeStackStatus(EdgeStackID, EdgeStackStatus) error
}
```

**Comment by deviantony** [comment]
@chiptus if you're talking about the API client in `http/client/client.go` it is currently designed to send requests to other agents inside the cluster (and as such, to talk to the agent API not the Portainer API).

You might want to create a PortainerClient here.

**Comment by chiptus** [comment]
ok

**Comment by deviantony** [comment]
Closed via #121
